tests/hypervisor/test_esx.py

```diff
@@ -853,3 +898,434 @@ def test_exclude_host_parents(self, virtwho, function_hypervisor, hypervisor_dat
+            server=hypervisor_data['hypervisor_server'],
+            admin_user=hypervisor_data['hypervisor_username'],
+            admin_passwd=hypervisor_data['hypervisor_password'],
+            client_server=hypervisor_data['ssh_ip'],
+            client_user=hypervisor_data['ssh_username'],
+            client_passwd=hypervisor_data['ssh_password']
+        )
+
+        try:
+            # change the vcenter cluster name to: virtwho/test
+            esx.cluster_name_set(host_ip, cluster_name, new_cluster_name)
+
+            # run virt-who service with the new cluster name
+            result = virtwho.run_service()
+            assert (result['error'] == 0
+                    and result['send'] == 1
+                    and result['thread'] == 1
+                    and f'"hypervisor.cluster": "{new_cluster_name}"' in result['log'])
+
+            # check the hyperivsor facts
+            if REGISTER == 'satellite':
+                host_id = satellite.host_id(host_name)
+                hypervisor_facts = satellite.facts_get(host_id)
+            else:
+                output = rhsm.info(host_name)
+                hypervisor_facts = output["facts"]["hypervisor.cluster"]
+            assert new_cluster_name in hypervisor_facts
+
+        finally:
+            # change back the vcenter cluster name
+            esx.cluster_name_set(host_ip, new_cluster_name, cluster_name)
+
+    @pytest.mark.tier2
+    @pytest.mark.notStage
+    def test_post_large_json_to_rhsm(self, register_data, ssh_host):
+        """
+        :title: virt-who: esx: post large json data to satellite server
+        :id: 8b14bb1f-7b92-483f-af35-7ec97a621436
+        :caseimportance: High
+        :tags: tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+            1. Create json data with 100 hypervisors, every hypervisor has 30 guests
+            2. Post the json data to satellite
+
+        :expectedresults:
+            2. Succeeded to post 600 hypervisors and 18000 guests to satellite
+
+        """
+        # create json data
+        local_file = os.path.join(TEMP_DIR, 'test.json')
+        json_file = "/root/test.json"
+        json_data = json_data_create(hypervisors=100, guests=30)
+        with open(local_file, "w") as f:
+            json.dump(json_data, f)
+        ssh_host.put_file(local_file, json_file)
+
+        # post json data
+        curl_header = (
+            '-H "accept:application/json,version=2" -H "content-type:application/json"'
+        )
+        curl_cert = "--cert /etc/pki/consumer/cert.pem --key /etc/pki/consumer/key.pem"
+        curl_json = f'-d @"{json_file}"'
+        curl_host = f"https://{register_data['server']}/rhsm/hypervisors"
+        cmd = f"curl -X POST -s -k {curl_header} {curl_cert} {curl_json} {curl_host}"
+
+        ret, output = ssh_host.runcmd(cmd)
+        if ret == 0 and "error" not in output:
+            logger.info("Succeeded to 600 hypervisors and 18000 guests to satellite")
```

Comment:
Succeeded to `post` 600 hypervisors and 18000 guests to satellite.
Small issue.

Reply:
Also Copied from the old framework, will update